tools/scan-build-py/libscanbuild/analyze.py

```diff
@@ -165,22 +165,28 @@ def generate_func_map_lines(fnmap_dir):
                 for line in in_file:
                     yield line
 
-    def write_global_map(ctudir, mangled_ast_pairs):
+    def write_global_map(ctudir, arch, mangled_ast_pairs):
         """ Write (mangled function name, ast file) pairs into final file. """
 
-        extern_fns_map_file = os.path.join(ctudir, CTU_FUNCTION_MAP_FILENAME)
+        extern_fns_map_file = os.path.join(ctudir, arch,
+                                           CTU_FUNCTION_MAP_FILENAME)
         with open(extern_fns_map_file, 'w') as out_file:
             for mangled_name, ast_file in mangled_ast_pairs:
                 out_file.write('%s %s\n' % (mangled_name, ast_file))
 
-    fnmap_dir = os.path.join(ctudir, CTU_TEMP_FNMAP_FOLDER)
+    triple_arches = glob.glob(os.path.join(ctudir, '*'))
+    for triple_path in triple_arches:
+        if os.path.isdir(triple_path):
+            triple_arch = os.path.basename(triple_path)
+            fnmap_dir = os.path.join(ctudir, triple_arch,
+                                     CTU_TEMP_FNMAP_FOLDER)
 
-    func_map_lines = generate_func_map_lines(fnmap_dir)
-    mangled_ast_pairs = create_global_ctu_function_map(func_map_lines)
-    write_global_map(ctudir, mangled_ast_pairs)
+            func_map_lines = generate_func_map_lines(fnmap_dir)
+            mangled_ast_pairs = create_global_ctu_function_map(func_map_lines)
```

Comment:
We no longer used mangled names so USR might better reflect what this map is.

Reply:
Isn't USR is just an other way of name mangling? For me, mangling seems to be a wider term independent of implementation. See https://en.wikipedia.org/wiki/Name_mangling